flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/org/apache/flink/cdc/connectors/mysql/source/MysqlPipelineNewlyAddedTableITCase.java

```diff
@@ -0,0 +1,579 @@
+        CollectSinkOperator<T> operator = (CollectSinkOperator<T>) sinkFactory.getOperator();
+        CollectResultIterator<T> iterator =
+                new CollectResultIterator<>(
+                        buffer, operator.getOperatorIdFuture(), accumulatorName, 0);
+        CollectStreamSink<T> sink = new CollectStreamSink<>(source, sinkFactory);
+        sink.name("Data stream collect sink");
+        env.addOperator(sink.getTransformation());
+        env.registerCollectIterator(iterator);
+        return iterator;
+    }
+
+    private StreamExecutionEnvironment getStreamExecutionEnvironment(
+            String finishedSavePointPath, int parallelism) {
+        Configuration configuration = new Configuration();
+        if (finishedSavePointPath != null) {
+            configuration.setString(SavepointConfigOptions.SAVEPOINT_PATH, finishedSavePointPath);
+        }
+        StreamExecutionEnvironment env =
+                StreamExecutionEnvironment.getExecutionEnvironment(configuration);
+        env.setParallelism(parallelism);
+        env.enableCheckpointing(500L);
+        env.setRestartStrategy(RestartStrategies.fixedDelayRestart(3, 1000L));
+        return env;
+    }
+
+    static class TestParam {
+        private final List<String> firstRoundInitTables;
+        private final List<String> firstRoundListenTables;
+        private final Integer firstRoundFetchSize;
+        private final List<String> secondRoundInitTables;
+        private final List<String> secondRoundListenTables;
+        private final Integer secondRoundFetchSize;
+
+        private TestParam(Builder builder) {
+            this.firstRoundInitTables = builder.firstRoundInitTables;
+            this.firstRoundListenTables = builder.firstRoundListenTables;
+            this.firstRoundFetchSize = builder.firstRoundFetchSize;
+            this.secondRoundInitTables = builder.secondRoundInitTables;
+            this.secondRoundListenTables = builder.secondRoundListenTables;
+            this.secondRoundFetchSize = builder.secondRoundFetchSize;
+        }
+
+        public static Builder newBuilder(
+                List<String> firstRoundListenTables,
+                Integer firstRoundFetchSize,
+                List<String> secondRoundListenTables,
+                Integer secondRoundFetchSize) {
+            return new Builder(
+                    firstRoundListenTables,
+                    firstRoundFetchSize,
+                    secondRoundListenTables,
+                    secondRoundFetchSize);
+        }
+
+        public static class Builder {
+            private List<String> firstRoundInitTables;
+            private final List<String> firstRoundListenTables;
+            private final Integer firstRoundFetchSize;
+
+            private List<String> secondRoundInitTables;
+            private final List<String> secondRoundListenTables;
+            private final Integer secondRoundFetchSize;
+
+            public Builder(
```

Comment:
Why firstRoundInitTables and firstRoundListenTables are passed by constructor , while firstRoundInitTables is set by setter?

Reply:
If we listen full database like 'db.\.*', we don't have to init any tables, because some tables have been initialized in the bofore() method(but I didn't implement this test case yet).